utils/log/loggers/fields.go

```diff
@@ -12,6 +13,10 @@ var (
 
 //LogFields contains all fields that have to be added to logs
 type LogFields map[string]interface{}
+type protectedLogFields struct {
+	content LogFields
+	mtx     sync.RWMutex
```

Comment:
As @ankurs suggested, `sync.Map` is indeed a better option.

According to https://pkg.go.dev/sync#Map, it optimizes for our use case:
> when the entry for a given key is only ever written once but read many times, as in caches that only grow